File: sync_footer.py
import os
import re

html_dir = r"f:\FBU\học tập\Javascript và lập trình web\jewelry-luxury\public\pages"
index_path = os.path.join(html_dir, 'index.html')

with open(index_path, 'r', encoding='utf-8') as f:
    index_content = f.read()

footer_match = re.search(r'(<footer class="footer">.*?</footer>)', index_content, flags=re.DOTALL)
if not footer_match:
    print("Could not find footer in index.html")
    exit(1)

good_footer = footer_match.group(1)

for filename in os.listdir(html_dir):
    if filename.endswith('.html') and filename != 'index.html':
        filepath = os.path.join(html_dir, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        new_content, count = re.subn(r'<footer[^>]*>.*?</footer>', good_footer, content, flags=re.DOTALL)
        
        if count > 0:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Synced footer in {filename}")
        else:
            print(f"No footer found to replace in {filename}")

File: sync_footer.js
const fs = require('fs');
const path = require('path');

const htmlDir = path.join(__dirname, 'public', 'pages');
const indexFile = path.join(htmlDir, 'index.html');

const indexContent = fs.readFileSync(indexFile, 'utf8');

// Match <footer class="footer"> ... </footer>
const footerMatch = indexContent.match(/(<footer class="footer">[\s\S]*?<\/footer>)/);
if (!footerMatch) {
    console.error("Footer not found in index.html");
    process.exit(1);
}

const goodFooter = footerMatch[1];
const files = fs.readdirSync(htmlDir);

for (const file of files) {
    if (file.endsWith('.html') && file !== 'index.html') {
        const filePath = path.join(htmlDir, file);
        const content = fs.readFileSync(filePath, 'utf8');
        
        // Replace existing footer or <footer class="footer"> block
        // some files might just have <footer>
        const newContent = content.replace(/<footer[^>]*>[\s\S]*?<\/footer>/g, goodFooter);
        
        if (newContent !== content) {
            fs.writeFileSync(filePath, newContent, 'utf8');
            console.log(`Synced footer in ${file}`);
        } else {
            console.log(`No footer found to replace in ${file}`);
        }
    }
}
